src/main/database_manager.cpp

```diff
@@ -71,7 +73,26 @@ void DatabaseManager::DetachDatabase(ClientContext &context, const string &name,
 	}
 }
 
-void DatabaseManager::InsertDatabasePath(const string &path, const string &name) {
+optional_ptr<AttachedDatabase> DatabaseManager::GetDatabaseFromPath(ClientContext &context, const string &path) {
+	auto database_list = GetDatabases(context);
+	for (auto &db_ref : database_list) {
+		auto &db = db_ref.get();
+		if (db.IsSystem()) {
+			continue;
+		}
+		auto &catalog = Catalog::GetCatalog(db);
+		if (catalog.InMemory()) {
+			continue;
+		}
+		auto db_path = catalog.GetDBPath();
+		if (StringUtil::CIEquals(path, db_path)) {
```

Comment:
I saw some performance improvements using a reversed comparison here, i.e., starting with the last string byte. But I am unsure how relevant it is now, as we're hopefully rarely executing this function. Or, could `GetDatabases(context);` return a map of paths-name here, and then we don't have to worst-case loop twice comparing each string? Just adding some performance thoughts... And we can use that map to get and return the specific database if there is a matching path.

Reply:
I think perf is not relevant here anymore so let's avoid optimizing